source/expectations/matchers/ut_be_within.tpb

```diff
@@ -0,0 +1,113 @@
+  constructor function ut_be_within(self in out nocopy ut_be_within, a_dist yminterval_unconstrained, a_is_pct number) return self as result is
+  begin
+    init(ut_data_value_yminterval(a_dist),a_is_pct);
+    return;
+  end;
+  
+  member procedure of_(self in ut_be_within, a_expected number) is
+    l_result ut_be_within := self;
+  begin 
+    l_result.expected := ut_data_value_number(a_expected);
+    l_result.expectation.to_(l_result );        
+  end;
+  
+  member procedure of_(self in ut_be_within, a_expected date) is
+    l_result ut_be_within := self;
+  begin 
+    l_result.expected := ut_data_value_date(a_expected);
+    l_result.expectation.to_(l_result );    
+  end;
+  
+  overriding member function run_matcher(self in out nocopy ut_be_within, a_actual ut_data_value) return boolean is
+    l_result boolean;
+  begin
+    if self.expected.data_type = a_actual.data_type then
+      if self.expected is of (ut_data_value_number) and self.is_pct = 0 then
+        l_result := abs((treat(self.expected as ut_data_value_number).data_value - treat(a_actual as ut_data_value_number).data_value)) <= 
+                    treat(self.dist as ut_data_value_number).data_value;
+      elsif self.expected is of (ut_data_value_number) and self.is_pct = 1 then
+        l_result := treat(self.dist as ut_data_value_number).data_value >= 
+                    (
+                     ((treat(self.expected as ut_data_value_number).data_value - treat(a_actual as ut_data_value_number).data_value ) * 100 ) /
+                    (treat(self.expected as ut_data_value_number).data_value)) ;      
+      elsif self.expected is of (ut_data_value_date) and self.dist is of ( ut_data_value_yminterval) then      
+        l_result := treat(a_actual as ut_data_value_date).data_value 
+                    between 
+                     (treat(self.expected as ut_data_value_date).data_value) - treat(self.dist as ut_data_value_yminterval).data_value
+                     and 
+                     (treat(self.expected as ut_data_value_date).data_value) + treat(self.dist as ut_data_value_yminterval).data_value;
+      elsif self.expected is of (ut_data_value_date) and self.dist is of ( ut_data_value_dsinterval) then      
+        l_result := treat(a_actual as ut_data_value_date).data_value 
+                    between 
+                      (treat(self.expected as ut_data_value_date).data_value) - treat(self.dist as ut_data_value_dsinterval).data_value
+                      and 
+                      (treat(self.expected as ut_data_value_date).data_value) + treat(self.dist as ut_data_value_dsinterval).data_value;
+      end if;
+    else
+      l_result := (self as ut_matcher).run_matcher(a_actual);
+    end if;
+    return l_result;
+  end;
+
+  overriding member function failure_message(a_actual ut_data_value) return varchar2 is
+    l_distance varchar2(32767);
+  begin
+    l_distance := case 
+                    when self.dist is of (ut_data_value_number) then
```

Comment:
I suspect that this CASE isn't needed.
I would expect the `to_string` method of the actual type to be invoked without need to use `treat`